components/edit-collective/actions/EmptyBalance.js

```diff
@@ -54,65 +55,127 @@ const EmptyBalance = ({ collective, LoggedInUser }) => {
+                LoggedInUser={LoggedInUser}
+                amount={collective.stats.balance}
+                currency={collective.currency}
+                confirmTransfer={confirmTransfer}
+                isTransferApproved={modal.isApproved}
               />
-            </StyledButton>
-          )}
-          <Modal show={modal.show} width="570px" onClose={closeModal}>
-            <ModalHeader onClose={closeModal}>
-              <FormattedMessage
-                id="collective.emptyBalance.header"
-                values={{ action: modal.type }}
-                defaultMessage={'{action} Balance'}
+            )}
+            {collective.stats.balance === 0 && (
+              <StyledButton disabled={true}>
+                <FormattedMessage
+                  id="SendMoneyToCollective.btn"
+                  defaultMessage="Send {amount} to {collective}"
+                  values={{
+                    amount: formatCurrency(0, collective.currency),
+                    collective: collective.host.hostCollective.name,
+                  }}
+                />
+              </StyledButton>
+            )}
+            <Modal show={modal.show} width="570px" onClose={closeModal}>
+              <ModalHeader onClose={closeModal}>
+                <FormattedMessage
+                  id="collective.emptyBalance.header"
+                  values={{ action: modal.type }}
+                  defaultMessage={'{action} Balance'}
+                />
+              </ModalHeader>
+              <ModalBody>
+                <P>
+                  <FormattedMessage
+                    id="collective.emptyBalance.body"
+                    values={{ collective: collective.host.hostCollective.name, action: modal.type.toLowerCase() }}
+                    defaultMessage={'Are you sure you want to {action} to {collective}?'}
+                  />
+                </P>
+              </ModalBody>
+              <ModalFooter>
+                <Container display="flex" justifyContent="flex-end">
+                  <StyledButton mx={20} onClick={() => setModal({ ...modal, show: false, isApproved: false })}>
+                    <FormattedMessage id="actions.cancel" defaultMessage={'Cancel'} />
+                  </StyledButton>
+                  <StyledButton
+                    buttonStyle="primary"
+                    data-cy="action"
+                    onClick={() => setModal({ ...modal, show: false, isApproved: true })}
+                  >
+                    <FormattedMessage id="confirm" defaultMessage={'Confirm'} />
+                  </StyledButton>
+                </Container>
+              </ModalFooter>
+            </Modal>
+          </Fragment>
+        )}
+      {console.log(collective)}
+      {(collective.type === CollectiveType.PROJECT || collective.type === CollectiveType.EVENT) &&
+        collective.parentCollective && (
+          <Fragment>
+            {collective.stats.balance > 0 && (
```

Comment:
There's a lot of code duplication here. In both cases, we're rendering the same `SendMoneyToCollectiveBtn`, `StyledButton` and `Modal` with 90% of the arguments being the same. Can we do something to improve that?

Reply:
+1 Better tweak the arguments based on the type.

Reply:
Done. 👍🏽